tests/jobs/rft/test_trajectory.py

```diff
@@ -0,0 +1,103 @@
+import itertools
+import unittest
+import pytest
+import argparse
+
+from semeio.jobs.rft.trajectory import TrajectoryPoint, Trajectory
+from semeio.jobs.rft.zonemap import ZoneMap
+from semeio.jobs.rft.utility import parse_trajectory_line, load_trajectory_points
+
+
+class TrajectoryPointTest(unittest.TestCase):
+    def test_load_from_line(self):
+
+        for line, expected_zone in zip(["0 1 2.2 3", "0 1 2.2 3 zone"], [None, "zone"]):
+            point = TrajectoryPoint(*parse_trajectory_line(line))
+            self.assertEqual(point.utm_x, 0)
+            self.assertEqual(point.utm_y, 1)
+            self.assertEqual(point.measured_depth, 2.2)
+            self.assertEqual(point.true_vertical_depth, 3)
+            self.assertEqual(point.zone, expected_zone)
+
+    def test_invalid_load_from_line(self):
```

Comment:
You know I like `parameterized`, but wouldn't this be more readable:
```suggestion
@pytest.mark.parametrize(
    "lines,expected_error",
    [
        ("1", "Trajectory data file not on correct format"),
        ("a b c", "Trajectory data file not on correct format"),
        ("a b c d e", "Error: Failed to extract data from line"),
    ],
)
def test_invalid_load_from_line(lines, expected_error):
    with pytest.raises(argparse.ArgumentTypeError) as e_info:
        TrajectoryPoint(*parse_trajectory_line(lines))
    assert expected_error in str(e_info.value)
```

Reply:
I do know you like your `parameterized`, but this can just as easily be written as:
```
def test_invalid_load_from_line():
    lines_errors = [
        ("1", "Trajectory data file not on correct format"),
        ("a b c", "Trajectory data file not on correct format"),
        ("a b c d e", "Error: Failed to extract data from line"),
    ]

    for (line, error) in lines_errors:
        with pytest.raises(argparse.ArgumentTypeError) as err_context:
            TrajectoryPoint(*parse_trajectory_line(line))
        assert error in err_context.value.args[0]
```
So you don't need the pytest fixture to change it into the other way of writing ;)

Some really like the `zip`, I'm a bit back and forth.. But I agree that it's easier to match which input provides which error, so I'll change it. And I'll even use `parametrize`!